src/main/java/com/google/sps/servlets/AuthenticateServlet.java

```diff
@@ -44,8 +46,38 @@ public void doGet(HttpServletRequest request, HttpServletResponse  response) thr
     response.setContentType("application/json;");
     Gson gson = new Gson();
 
-    GivrUser user = GivrUser.getLoggedInUser();
+    GivrUser user = GivrUser.getCurrentLoggedInUser();
 
+    Map<String, String> propertyNamesAndValuesToUpdate = new HashMap<String, String>();
+
+    boolean doesUserIdExistInDatastore = GivrUser.checkIfUserWithPropertyExists("userId", user.getUserId());
+    if (!doesUserIdExistInDatastore) {
+      boolean doesUserEmailExistInDatastore = GivrUser.checkIfUserWithPropertyExists("userEmail", user.getUserEmail());
+      if (doesUserEmailExistInDatastore) {
```

Comment:
if (!userInDatastore) {
   MaybeUpdateUserByEmailInDatastore(user.email);
} else {
   MaybeUpdateEmailAddressOfUserIdInDatastore(user);  // passing in the givruser
}